core/core/test/serverspec/spec/master/master_spec.rb

```diff
@@ -1,28 +1,158 @@
 require 'spec_helper'
+require "base64"
 
-describe service('kubelet') do
-  it { should be_enabled }
-  it { should be_running }
+
+describe 'Check if kubelet service is up' do
+  describe service('kubelet') do
+    it { should be_enabled }
+    it { should be_running }
+  end
+end
+
+describe 'Check if kube-scheduler is up' do
+  describe process("kube-scheduler") do
+    it { should be_enabled }
+    it { should be_running }
+  end
+end
+
+describe 'Check if kube-scheduler is up - 2nd test' do
+  describe command('ps -C kube-scheduler --no-headers') do
+    its(:stdout) { should match /kube-scheduler/ }
+    its(:exit_status) { should eq 0 }
+  end
 end
 
-describe service('kube-controller') do
+describe 'Check if kube-controller is up' do
+  describe process("kube-controller") do
     it { should be_enabled }
     it { should be_running }
   end
+end
+
+describe 'Check if kube-controller is up - 2nd test' do
+  describe command('ps -C kube-controller --no-headers') do
+    its(:stdout) { should match /kube-controller/ }
+    its(:exit_status) { should eq 0 }
+  end
+end
 
-describe service('kube-apiserver') do
+describe 'Check if kube-apiserver is up' do
+  describe process("kube-apiserver") do
     it { should be_enabled }
     it { should be_running }
   end
+end
+
+describe 'Check if kube-apiserver is up - 2nd test' do
+  describe command('ps -C kube-apiserver --no-headers') do
+    its(:stdout) { should match /kube-apiserver/ }
+    its(:exit_status) { should eq 0 }
+  end
+end
+
+describe 'Check if there are any pods that have status other than Running' do
+  describe command('kubectl get pods --all-namespaces --field-selector=status.phase!=Running') do
+    its(:stdout) { should match /^$/ }
+    its(:stderr) { should match /No resources found./ }
+  end
+end  
+
+describe 'Check if the number of master nodes is the same as indicated in the inventory file' do
+  describe command('kubectl get nodes --selector=node-role.kubernetes.io/master --no-headers | wc -l | tr -d "\n"') do
+    its(:stdout) { should eq count_inventory_roles("master").to_s }
```

Comment:
wouldn't it be better to convert kubectl output to integer and compare numbers instead of converting count_inventory to string and compare strings? 

same thing with line 69

Reply:
fixed in 5683acf